modules/report/includes/commerce_pos_report.pages.inc

```diff
@@ -1005,3 +1004,300 @@ function commerce_pos_report_build_journal_role_table(array $filters = array())
+    ->extend('PagerDefault')
+    ->extend('TableSort');
+
+  $subQuery->fields('cpc', array('cashier_id'));
+  $subQuery->limit($filters['results_per_page']);
+
+  $limit_rows_result = $subQuery->execute();
+  if ($limit_rows_result->rowCount() > 0) {
+    foreach ($limit_rows_result as $fetchCashier) {
+      $cashiers_id[] = $fetchCashier->cashier_id;
+    }
+  }
+  $query->condition('cashier.cashier_id', $cashiers_id, 'IN');
+
+  // Set up our filters/conditions.
+  $query->orderByHeader($header);
+  $result = $query->execute();
+
+  $rows = array();
+  $unit_per_transaction = 0;
+  $net_sales = 0;
+  $totals = array(
+    'net_sales' => 0,
+    'transactions' => 0,
+    'item_sold' => 0,
+    'unit_per_transaction' => 0,
+    'dollar_per_transaction' => 0,
+  );
+
+  if ($result->rowCount() > 0) {
+    $report = kpi_process_data($result);
+    foreach ($report as $cashier_transactions) {
+      $table_row              = array();
+      $transactions           = $cashier_transactions['cashier']['transactions'];
+      $currency_code          = $cashier_transactions['cashier']['currency_code'];
+      $net_sales              = $cashier_transactions['transaction']['net_sales'];
+      $net_sold_items         = ($cashier_transactions['transaction']['purchase_count'] - $cashier_transactions['transaction']['return_count']);
+      $unit_per_transaction   = round(($net_sold_items / $transactions), 2);
+      $dollar_per_transaction = ($net_sales / $transactions);
+      $table_row[]            = $cashier_transactions['cashier']['id'];
+      $table_row[]            = $cashier_transactions['cashier']['name'];
+      $table_row[]            = commerce_currency_format($net_sales, $currency_code);
+      $table_row[]            = $transactions;
+      $table_row[]            = $net_sold_items;
+      // Unit per Trans (UPT)
+      $table_row[] = $unit_per_transaction;
+      $table_row[] = commerce_currency_format($dollar_per_transaction, $currency_code);
+      // Table feet with total.
+      $totals['net_sales'] += $net_sales;
+      $totals['transactions'] += $transactions;
+      $totals['item_sold'] += $net_sold_items;
+      $totals['unit_per_transaction'] += $unit_per_transaction;
+      $totals['dollar_per_transaction'] += $dollar_per_transaction;
+      $rows[] = array(
+        'data' => $table_row,
+        'class' => '',
+      );
+    }
+
+    $rows[] = array(
+      '',
+      '',
+      commerce_currency_format($totals['net_sales'], $currency_code),
+      $totals['transactions'],
+      $totals['item_sold'],
+      $totals['unit_per_transaction'],
+      commerce_currency_format($totals['dollar_per_transaction'], $currency_code),
+    );
+
+    $table = array(
+      '#theme' => 'table',
+      '#rows' => $rows,
+      '#header' => $header,
+      '#attached' => array(
+        'js' => array(
+          drupal_get_path('module', 'commerce_pos_report') . '/js/commerce_pos_report.journal_role.js',
```

Comment:
Done. I was looking to save time by not involving code changes at old as well as new function as not just javascript file even css is dependent of id selector , I have recreated the css with a class rather than id so its reusable and we can have different name for css classes